modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockService.scala

```diff
@@ -41,7 +42,7 @@ object BlockService {
 
       def accept(signedBlock: Signed[B]): F[Unit] =
         signedBlock.toHashed.flatMap { hashedBlock =>
-          EitherT(blockAcceptanceManager.acceptBlock(signedBlock, context))
+          EitherT(blockAcceptanceManager.acceptBlock(signedBlock, context, transactionStorage.getInitialTxRef))
```

Comment:
I would use a single implementation so the one defined here a few lines lower, it will get especially easy to use it if the `getInitialTxRef` won't require `F`

Reply:
done :+1: